lib/tools/system-calls.js

```diff
@@ -546,13 +546,20 @@ systemCallMethods.killAllEmulators = async function () {
  * @param {string} avdName - The name of the emulator to be killed.
  */
 systemCallMethods.killEmulator = async function (avdName) {
-  log.debug(`killing avd '${avdName}'`);
-  let device = await this.getRunningAVD(avdName);
-  if (device) {
-    await this.adbExec(['emu', 'kill']);
-    log.info(`successfully killed emulator '${avdName}'`);
+  if (util.hasValue(avdName)) {
+    log.debug(`killing avd '${avdName}'`);
+    let device = await this.getRunningAVD(avdName);
+    if (device) {
+      await this.adbExec(['emu', 'kill']);
+      log.info(`successfully killed emulator '${avdName}'`);
+    } else {
+      log.info(`no avd with name '${avdName}' running. skipping kill step.`);
```

Comment:
We could probably return false or true based on the fact that emulator has been killed or not